File: elixir/lib/polyglot/reverse.ex
# https://github.com/joncalhoun/algorithmswithgo.com/tree/master/module01#07---reverse-a-string-code

defmodule Polyglot.Reverse do
  def pattern_match(""), do: ""

  def pattern_match(word) do
    {head, tail} = String.next_codepoint(word)
    pattern_match(tail) <> head
  end

  def std(word) do
    String.reverse(word)
  end
end

File: rust/src/reverse.rs
// https://github.com/joncalhoun/algorithmswithgo.com/tree/master/module01#07---reverse-a-string-code

pub fn iterate(word: &'static str) -> String {
    let mut res = String::from("");
    for w in word.chars() {
        res.insert(0, w)
    }
    res
}

pub fn std(word: &'static str) -> String {
    word.chars().rev().collect()
}
